packages/app-desktop/main.js

```diff
@@ -36,7 +36,17 @@ const env = envFromArgs(process.argv);
 const profilePath = profileFromArgs(process.argv);
 const isDebugMode = !!process.argv && process.argv.indexOf('--debug') >= 0;
 
-const wrapper = new ElectronAppWrapper(electronApp, env, profilePath, isDebugMode);
+if (env === 'dev' && process.platform === 'win32') {
+	electronApp.setAsDefaultProtocolClient('joplin', process.execPath, [
+		resolve(process.argv[1])
+	]);
+} else {
+	electronApp.setAsDefaultProtocolClient('joplin');
+}
+
+const initialUrl = process.argv.find((arg) => arg.startsWith('joplin://'));
```

Comment:
Should start with joplin://x-callback-url/. Also maybe create a helper function to check if a url is valid?